lib/charms/observability_libs/v0/cert_manager.py

```diff
@@ -345,4 +345,8 @@ def _on_all_certificates_invalidated(self, event: AllCertificatesInvalidatedEven
     def _on_certificates_relation_broken(self, event: RelationBrokenEvent) -> None:
         """Clear the certificates data when removing the relation."""
         if self._peer_relation:
-            pass # TODO clear/delete the peer relation
+            if private_key := self._private_key:  # preserve the private key
+                self._peer_relation.data[self.charm.unit].clear()
```

Comment:
@lucabello shouldn't we clear `event.unit`  instead of `self.charm.unit`?

Reply:
you can't clear the databag of another unit, only your own

Reply:
Ok so not `clear()`. How about `del` or `pop`?
We don't need to clear the remote data bag in the juju sense, just in the ops data structure, for this particular hook.
The following line,
```
self._peer_relation.data[self.charm.unit].update({"private_key": private_key})
```
is quite brittle: we'd rather not manually recreate reldata to schema.

Reply:
Hmm:
```
TypeError: 'RelationData' object does not support item deletion
```
```
AttributeError: 'RelationData' object has no attribute 'pop'
```